index.html

```diff
@@ -705,31 +710,65 @@
                                 var inputPort, outputPort, controlPort, commandPort;
                                 if (cellvar.model.in.height == null) {
                                     inputPort = 0;
-                                } else {
+                                }
+                                else {
                                     inputPort = cellvar.model.in.height;
                                 }
                                 if (cellvar.model.out.height == null) {
                                     outputPort = 0;
-                                } else {
+                                }
+                                else {
                                     outputPort = cellvar.model.out.height;
                                 }
                                 if (cellvar.model.evtin.height == null) {
                                     controlPort = 0;
-                                } else {
+                                }
+                                else {
                                     controlPort = cellvar.model.evtin.height;
                                 }
                                 if (cellvar.model.evtout.height == null) {
                                     commandPort = 0;
-                                } else {
+                                }
+                                else {
                                     commandPort = cellvar.model.evtout.height;
                                 }
+
+                                var flip = false, mirror = false;
+                                if(cell.style.indexOf('stencilFlipV') == -1){
```

Comment:
Ya your styleToObject was not very complete. I have put in a condition.
Check it.

On 13 July 2016 at 00:03, Jitesh Kumar Jha <notifications@github.com> wrote:

> In index.html
> <https://github.com/FOSSEE/xcos-on-web/pull/170#discussion_r70495447>:
>
> >                                      commandPort = cellvar.model.evtout.height;
> >                                  }
> > +
> > +                                var flip = false, mirror = false;
> > +                                if(cell.style.indexOf('stencilFlipV') == -1){
>
> The styleToObject function works for mxCells having initial style (which
> is imported directly from the stylesheet) and along with the styles added
> during user operation to the mxCell.
>
> For example, a BIGSUM_f block has initial style as 'BIGSUM_F'. If we apply
> fill colour to that block's style, the resultant style will be 'BIGSUM_f;
> strokeColor=#someColor;'
>
> What styleToObject function does it, that it returns the object format of
> the style which is in string format, in our case, the result will be
> something like --
>
> {
> default: BIGSUM_f,
> strokeColor: #somecolor
> }
>
> NOTE: There exists objectToStyle function, which converts style Object
> format to style String format.
> Suppose we would like to set some style property to the resultant object
> we received from styleToObject function, we can do it by --
>
>
> // Assuming 'style' variable as style of a block in the string format
> var styleObject = styleToObject(style);
> style['fillColor'] = 'someColor';
> style = objectToStyle(style);
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/FOSSEE/xcos-on-web/pull/170/files/478f4d2ac25195ddd948d713fd975e973ebd7c3a#r70495447>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALzP2tL6UQ1l4eHSkm8WT-psDmQFgDi8ks5qU93jgaJpZM4JKJM6>
> .
>